server/src/main/java/org/apache/iotdb/db/mpp/plan/planner/OperatorTreeGenerator.java

```diff
@@ -2208,25 +2209,71 @@ public Operator visitPathsUsingTemplateScan(
             node.getPathPatternList(), node.getTemplateId()));
   }
 
-  private List<Operator> dealWithConsumeAllChildrenPipelineBreaker(
+  public List<Operator> dealWithConsumeAllChildrenPipelineBreaker(
       PlanNode node, LocalExecutionPlanContext context) {
     // children after pipelining
-    List<Operator> children = new ArrayList<>();
+    List<Operator> parentPipelineChildren = new ArrayList<>();
     int finalExchangeNum = context.getExchangeSumNum();
-    for (PlanNode childSource : node.getChildren()) {
-      // Create pipelines for children
-      LocalExecutionPlanContext subContext = context.createSubContext();
-      Operator childOperation = childSource.accept(this, subContext);
-      // If the child belongs to another fragment instance, we don't create pipeline for it
-      if (childOperation instanceof ExchangeOperator) {
-        children.add(childOperation);
-        finalExchangeNum += 1;
-      } else {
+    if (context.getDegreeOfParallelism() == 1) {
+      // If dop = 1, we don't create extra pipeline
+      for (PlanNode localChild : node.getChildren()) {
+        Operator childOperation = localChild.accept(this, context);
+        parentPipelineChildren.add(childOperation);
+      }
+    } else {
+      // Keep it since we may change the structure of origin children nodes
+      List<PlanNode> afterwardsNodes = new ArrayList<>();
+      // 1. Separate localChildren from all nodes
+      int exchangeNodeIndex = node.getChildren().size();
+      for (int i = 0; i < node.getChildren().size(); i++) {
+        // We do this based on that exchangeNode should appear at last
```

Comment:
I'm not very sure whether it's right. Need to be confirmed by @xingtanzjr @Wei-hao-Li

Reply:
<img width="1920" alt="image" src="https://user-images.githubusercontent.com/16079446/218654694-b39aadfe-285c-4b7e-8c05-c432e3e336ad.png">
I checked the code in `ExchangeNodeAdder`, you can find that the root data region is not always the first child's. More details can been seen in `calculateDataRegionByChildren` method.